emcc.py

```diff
@@ -2972,13 +2974,15 @@ def check_bad_eq(arg):
       settings_changes.append('PTHREADS_PROFILING=1')
       newargs[i] = ''
     elif newargs[i] == '-fno-exceptions':
-      settings_changes.append('DISABLE_EXCEPTION_CATCHING=1')
-      settings_changes.append('DISABLE_EXCEPTION_THROWING=1')
+      shared.Settings.DISABLE_EXCEPTION_CATCHING = 1
+      shared.Settings.DISABLE_EXCEPTION_THROWING = 1
+      shared.Settings.EXCEPTION_HANDLING = 0
     elif newargs[i] == '-fexceptions':
-      settings_changes.append('DISABLE_EXCEPTION_CATCHING=0')
-      settings_changes.append('DISABLE_EXCEPTION_THROWING=0')
+      eh_enabled = True
+    elif newargs[i] == '-fwasm-exceptions':
+      wasm_eh_enabled = True
```

Comment:
This line could be replaced by
```python
      shared.Settings.DISABLE_EXCEPTION_CATCHING = 1
      shared.Settings.DISABLE_EXCEPTION_THROWING = 1
      shared.Settings.EXCEPTION_HANDLING = 1
```
and then we can remove the unnecessary `wasm_eh_enabled` etc. locals.

Reply:
The reason I use this local is, we don't know whether we should set 
```py
      shared.Settings.EXCEPTION_HANDLING = 0
      shared.Settings.DISABLE_EXCEPTION_THROWING = 0
      shared.Settings.DISABLE_EXCEPTION_CATCHING = 0
```
these until we scan the whole thing. You're right that the moment we have `-fwasm-exceptions`, we can set those three lines you mentioned. But whether we should turn on Emscripten EH is determined after we scan all the flags. We can't set them the moment we see `-fexceptions`, because there might be an additional `-fwasm-exceptions`. That's the reason I defer configuration of all the settings to the end.